ts/src/main/scala/ch/epfl/ts/data/StrategyParameters.scala

```diff
@@ -0,0 +1,183 @@
+package ch.epfl.ts.data
+
+import ch.epfl.ts.data.Currency.Currency
+import scala.reflect.ClassTag
+
+
+/**
+ * Class behaving much like a map, designed to hold values
+ * for a trading strategy's parameters.
+ * Each parameter checks the validity of its value on instantiation.
+ * 
+ * It can be constructed with an arbitrary number of parameters.
+ */
+class StrategyParameters(params: (String, Parameter[_])*) {
+  type Key = String
+  val parameters = params.toMap
+  
+  /**
+   * Similar to what a map's `get` would do.
+   * We also perform type checking.
+   */
+  def get[T : ClassTag](key: Key): Option[T] = {
+    parameters.get(key) match {
+      case Some(p) => p.get() match {
```

Comment:
The idea is:
- The strategy demands a parameter under name with a certain type (key -> parameter). In this function, `T` gives the type of the value held by the parameter, `key` gives the name (e.g. `"tradingVolume"` or such).